empress/recon_vis/recon_viewer.py

```diff
@@ -96,83 +234,387 @@ def render_parasite_helper(fig, node, recon, host_lookup, show_internal_labels,
+
+    while host_node.layout.row != stop_row and host_node.parent_node:
+        parent_node = host_node.parent_node
+        if parent_node.layout.row < host_node.layout.row:
+            v_track = parent_node.iter_track(tree.Track.UPPER_VERTICAL)
+        else:
+            v_track = parent_node.iter_track(tree.Track.LOWER_VERTICAL)
+            while v_track < parent_node.layout.upper_v_track:
+                v_track = parent_node.iter_track(tree.Track.LOWER_VERTICAL)
+        h_track = parent_node.iter_track(tree.Track.HORIZONTAL)
+        offset = parent_node.layout.offset
+
+        sub_parent_pos = plot_tools.Position(parent_node.layout.x - (offset * v_track), \
+            parent_node.layout.y + (offset * h_track))
+
+        _render_loss_branch(sub_parent_pos, current_pos, fig)
+
+        host_node = parent_node
+        current_pos = sub_parent_pos
+    
+    node_pos = plot_tools.Position(node.layout.x, node.layout.y)
+    mid_pos = plot_tools.Position(node_pos.x, current_pos.y)
+
+    fig.line(node_pos, mid_pos, render_settings.PARASITE_EDGE_COLOR)
+    fig.line(mid_pos, current_pos, render_settings.PARASITE_EDGE_COLOR)
+
+
+def _event_color_shape(event: recon.Event):
+    """
+    Gives the color and shape for drawing event, depending on event type
+    :param event: Event object
+    :return A tuple with the color and shape of an event
+    """
+    if event.event_type is recon.EventType.TIPTIP:
+        return render_settings.LEAF_NODE_COLOR, render_settings.LEAF_NODE_SHAPE
+    if event.event_type is recon.EventType.COSPECIATION:
+        return render_settings.COSPECIATION_NODE_COLOR, render_settings.COSPECIATION_NODE_SHAPE
+    if event.event_type is recon.EventType.DUPLICATION:
+        return render_settings.DUPLICATION_NODE_COLOR, render_settings.DUPLICATION_NODE_SHAPE
+    if event.event_type is recon.EventType.TRANSFER:
+        return render_settings.TRANSFER_NODE_COLOR, render_settings.TRANSFER_NODE_SHAPE
+    return None, None 
+
+
+def _set_host_node_layout(host_tree: tree.Tree):
     """
     Sets the logicalRow and logicalCol values of each Node in host_tree.
     Assumes that each host Node has its order set already and this function
     uses those values and structure of the tree to set the logicalRow and logicalCol
     :param host_tree:  A Tree object representing the host tree
     :return None
     """
-    # sets logical row values for leaves in the order they appear in the list of host tree leaves
+    # Sets logical row values for leaves in the order they appear in the list of host tree leaves
     logical_row_counter = 0
     for leaf in host_tree.leaf_list():
         leaf.layout.row = logical_row_counter
-        leaf.layout.x = leaf.layout.col  # This can be scaled if desired
-        leaf.layout.y = leaf.layout.row  # This can be scaled if desired
+        leaf.layout.x=leaf.layout.col           # This can be scaled if desired
```

Comment:
Please add the space between `=` back and use two spaces before `#`

Reply:
Fixed